rust/noosphere-cli/tests/peer_to_peer.rs

```diff
@@ -222,31 +250,40 @@ async fn gateway_publishes_and_resolves_petnames_configured_by_the_client() {
         third_party_gateway_task.abort();
     });
     client_task.await.unwrap();
+
+    // Restart gateway and name system, ensuring republishing occurs
+    let (ns_task, ns_url) = start_name_system_server(&ipfs_url).await.unwrap();
```

Comment:
Clean up warnings here

See what we do here: https://github.com/subconsciousnetwork/noosphere/blob/6c6b8d92db859ccd1dc8bd6055671b7db82596e3/rust/noosphere-cli/tests/peer_to_peer.rs#L248-L250